File: level2/짝지어_제거하기.cpp
#include <iostream>
#include <string>
#include <stack>

using namespace std;

int solution(string s) {
    int pass = false;
    stack<char> st;

    if(s.size()%2 ==1)
        return 0;

    for(auto c : s){
        if(st.empty() || st.top() != c)
            st.push(c);
        else
            st.pop();
    }

    if(st.empty())
        return 1;
    return 0;
}

File: JavaScript/level2/짝지어_제거하기.js
function solution(s)
{
    let array = [];
    
    if(s.length%2 === 1)
        return 0;
    
    for(let i of s){
        if(array.length === 0 || array[array.length-1] !== i)
            array.push(i);
        else
            array.pop();
    }

    if(array.length === 0)
        return 1;
    return 0;
}